lib/single-new-eks-opensource-observability-construct/amp-rules-configurator-addon.ts

```diff
@@ -0,0 +1,60 @@
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+import { loadYaml, readYamlDocument } from "@aws-quickstart/eks-blueprints/dist/utils";
+import { Construct } from 'constructs';
+import { KubernetesManifest } from "aws-cdk-lib/aws-eks";
+
+/**
+ * Configuration options for add-on
+ */
+export interface AmpRulesConfiguratorAddOnProps {
+    /** 
+     * AMP workspace ID.
+     */
+    ampWorkspaceId: string;
+
+    /** 
+     * Namespace where the RuleGroupsNamespace objects will be deployed in
+     * @default "ack-system"
+     */
+    namespace?: string;
+}
+
+/**
+ * Default props
+ */
+const defaultProps = {
+    namespace: "ack-system"
+};
+
+/**
+ * Implementation of Rules configuration for AMP
+ */
+export class AmpRulesConfiguratorAddOn implements blueprints.ClusterAddOn {
+
+    readonly ampRulesConfiguratorAddOn: AmpRulesConfiguratorAddOnProps;
+
+    constructor(props: AmpRulesConfiguratorAddOnProps) {
+        this.ampRulesConfiguratorAddOn = { ...defaultProps, ...props };
+    }
+
```

Comment:
`@dependable` is required for dependent addons like AmpAddon etc. If not this might run ahead of time.